components/giftbox.vue

```diff
@@ -0,0 +1,280 @@
+<template>
+  <div class="giftbox-wrapper">
+    <div class="giftbox-button" @click="showModal">
+      <fa icon="gift" class="giftbox-icon" />
+    </div>
+    <b-modal
+      ref="giftbox-modal"
+      class="giftbox-modal"
+      size="xl"
+      title="Nieuw in onze API en widgets"
+      :scrollable="false"
+      hide-footer
+    >
+      <div class="features-wrapper">
+        <div class="features-list">
+          <div
+            v-for="feature in features"
+            :key="feature.uid"
+            :class="[
+              'feature-title-block',
+              { selected: feature.uid === selectedFeature.uid },
+              { seen: seenFeatures.has(feature.uid) },
+            ]"
+            @click="switchFeature(feature.uid)"
+          >
+            <div class="notification-icon-wrapper">
+              <fa
+                v-if="!seenFeatures.has(feature.uid)"
+                icon="eye"
+                class="eye-icon"
+              />
+              <fa v-else icon="eye-slash" class="eye-icon" />
+            </div>
+            <p>{{ feature.title }}</p>
+          </div>
+        </div>
+        <div v-if="selectedFeature" class="features-detail">
+          <div class="title-calltoaction-container">
+            <h6>{{ selectedFeature.title }}</h6>
+            <a
+              :href="selectedFeature.callToAction"
+              :alt="selectedFeature.callToActionLabel"
+              target="_blank"
+              class="btn-call-to-action"
+            >
+              {{ selectedFeature.callToActionLabel }}
+            </a>
+          </div>
+          <a
+            v-if="selectedFeature.image"
+            :href="selectedFeature.image"
+            target="_blank"
+          >
+            <img
+              :src="selectedFeature.image"
+              :alt="selectedFeature.title"
+              class="feature-image"
+            />
+          </a>
+          <!-- eslint-disable-next-line vue/no-v-html -->
+          <div class="feature-content" v-html="selectedFeature.body" />
+        </div>
+        <div v-else class="features-detail">
+          <p>Geen nieuwe features te melden.</p>
+        </div>
+      </div>
+    </b-modal>
+  </div>
+</template>
+
+<script>
+  export default {
+    name: 'Giftbox',
+    data: () => ({
+      loading: true,
+      features: [],
+      selectedFeature: undefined,
+      seenFeatures: new Set(),
+    }),
+    created() {
+      this.fetchFeatures().then(res => {
+        this.features = res;
+        if (this.features.length > 0) {
+          this.selectedFeature = this.features[0];
+          this.seenFeatures.add(this.selectedFeature.uid);
+        }
+      });  
+    },
+    methods: {
+      async fetchFeatures() {
+        this.loading = true;
+        const response = await fetch(
+          'https://www.publiq.be/uitdatabank/articles.json',
+        );
+        const data = await response.json();
```

Comment:
I thin you can do something like this
```suggestion
        const { data } = await response.json();
        return data;
```

Reply:
Yes, that should work. Here's more info on that syntax @simonpubliq https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring

Also more info on creating objects with variables and their names as properties and values that I showed you in our call earlier today: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#New_notations_in_ECMAScript_2015

Reply:
But note that we don't have Babel in the AngularJS app, so don't use this new syntax there.

Reply:
> But note that we don't have Babel in the AngularJS app, so don't use this new syntax there.

Shouldn't be an issue since the component lives inside the vue app?

Reply:
Yes it's no issue here, that's why I said don't use it _there_ 😉 But yeah it was a bit unclear at first glance